### PR TITLE
Split example app into OAuth and OAuth2

### PR DESCRIPTION
**Pull request by bf4**
I was having trouble running the example app because I don't have
OAuth creds and omniauth was yelling at me that there was no `env['rack.session']` even though there was a cookie `rack.session` (`OmniAuth::NoSessionError (You must provide a session to use OmniAuth.)`) and I couldn't get it to let the app load.

There are some pretty dramatic changes in here. Some of it is a bit hacky, like how I'm sharing common setup between the apps, but I don't think it's that hard to follow.

I like removing from the gemspec any files not required to run the tests. I'd remove awesome_print too, or put in the Gemfile, but that's outside of my scope.

The first commit will be the noisest as it splits example/app.rb into example/oauth.rb and example/oauth2.rb.
The subsequent commits should be easier to read.

Don't merge this until you've run it yourself per instrutions in the README, esp since I can't test the legacy OAuth one.

### CODE REVIEW

**Comment by minimul** [comment]
Tested against OAuth1. Works great.

If you could make this [change](https://github.com/bf4/qbo_api/compare/split_auth_paths...minimul:pr/83?expand=1) I will merge.

**Comment by bf4** [comment]
@minimul Did the instructions in the README make sense too?

I'll make that change and then if you want to merge this, follow up with a PR splitting out the vcr recordings by creds being used.

Something like

```diff
 def creds
+  case creds_type
+  when :oauth2_creds then oauth2_creds
+  when :oauth1_creds then oauth1_creds
+  else fail NotImplementedError, "No creds configure for type: '#{creds_type.inspect}'"
+  end
+end
+
+def creds_type
   if ENV['QBO_API_OAUTH2_ACCESS_TOKEN']
-    oauth2_creds
+    :oauth2_creds
   else
-    oauth1_creds
+    :oauth1_creds
   end
 end
```

and

```diff
def use_cassette(name, options={})
  # Set VCR_RECORD=once to re_record
  record_option = ENV.fetch("VCR_RECORD") { "none" }.to_sym
  options = {record: record_option }.merge!(options)
+  name = "#{creds_type}/#{name}" unless name =~ /oauth/
  VCR.use_cassette(name, options) do |cassette|
```

**Comment by bf4** [comment]
Also, since the example apps are now run separately, I'd like to change the paths of the oauth2 one from `get /oauth2` to `get /` because we can. Again, I thought that'd be better for a followup pr

**Comment by minimul** [comment]
@bf4 Yep. The README looks great.

re: PR splitting the creds. Good idea.
